spec/build/addons/jwt_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'spec_helper'
+
+describe Travis::Build::Addons::Jwt, :sexp do
+  let(:script) { stub('script') }
+  let(:data)   { payload_for(:push, :ruby, config: { addons: { jwt: config } }) }
+  let(:sh)     { Travis::Shell::Builder.new }
+  let(:addon)  { described_class.new(script, sh, Travis::Build::Data.new(data), config) }
+  subject      { sh.to_sexp }
+  before       { Time.stubs(:now).returns(Time.mktime(1970,1,1)) }
+  before       { addon.before_before_script }
+
+  describe 'jwt token, one secret' do
+    let(:config) { 'MY_ACCESS_KEY=987654321' }
+    it "should work" do
+      subject.should include_sexp [:echo, 'Initializing JWT', ansi: :yellow]
+      expected = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MH0.vHZdjuxQt6CdFmWPEJLAsVbI_KCxbwJVbgT7jaTxkK4"
+      subject.should include_sexp [:export, ['MY_ACCESS_KEY', expected]]
+    end
+  end
+
+  describe 'jwt token, several secrets' do
+    let(:config) { {
+      secret1: 'MY_ACCESS_KEY_1=123456789',
+      secret2: 'MY_ACCESS_KEY_2=ABCDEF'
+    } }
+    it "should work" do
+      subject.should include_sexp [:echo, 'Initializing JWT', ansi: :yellow]
+      expected1 = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MH0.iO-I8CNXhQlgw_dL8R9CDaDPplSC3yf9J399Uumn6CE"
+      subject.should include_sexp [:export, ['MY_ACCESS_KEY_1', expected1]]
+      expected2 = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MH0.p1B9HnYCh-z5Igek5tr_QVPb1zV1ucwcPZNvY039WKg"
```

Comment:
Are these expected values perhaps platform-dependent?

On my Mac, I get this failure:

```
  2) Travis::Build::Addons::Jwt jwt token, several secrets should include sexp :export and ["MY_ACCESS_KEY_1", "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MH0.iO-I8CNXhQlgw_dL8R9CDaDPplSC3yf9J399Uumn6CE"]
     Failure/Error: should include_sexp [:export, ['MY_ACCESS_KEY_1', expected1]]
       Expected the generated sexp to include:

       [:export, ["MY_ACCESS_KEY_1", "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MH0.iO-I8CNXhQlgw_dL8R9CDaDPplSC3yf9J399Uumn6CE"]]

       Found the following similar nodes:

       [:export, ["MY_ACCESS_KEY_1", "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MTgwMDB9.6s1WY-acH-kd9B0VsMjnzwrTgN2X9lEBbNfDDLfSG8M"]]
       [:export, ["MY_ACCESS_KEY_2", "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzbHVnIjoidHJhdmlzLWNpL3RyYXZpcy1jaSIsInB1bGwtcmVxdWVzdCI6IiIsImlhdCI6MTgwMDB9.FrkyJ7CI8xpAcVB8vea6NV2aLKG3ElN3R3J1nVbkgWQ"]]
     # ./spec/build/addons/jwt_spec.rb:29:in `block (3 levels) in <top (required)>'
```

which we do not get on Travis CI.

If so, this is a poor spec in my opinion.

I am not sure, though, how we might test these otherwise. Thoughts?